Documents new Azure deployment methods for 8.12 (#4551)
elastic/security-docs: docs/cloud-native-security/cspm-get-started-azure.asciidoc

[[cspm-setup-azure]]
== Set up CSPM for Azure

> You can set up CSPM for Azure either by by enroling an Azure organization containing multiple subscriptions, or by enroling a single subscription. Either way, first you will add the CSPM integration, then enable cloud account access.

Reviewer: Let's say "Azure organization (root management group)" instead of just Azure orgnization.

Reviewer 2: or just "Azure organization (management group)" since non-root management groups work as well

[discrete]
[[cspm-set-up-ARM]]
== ARM template setup (recommended)

NOTE: If you are deploying to an Azure organization, you need the following permissions: `Microsoft.Resources/deployments/*`, `Microsoft.Authorization/roleAssignments/write`. You also need to https://learn.microsoft.com/en-us/azure/role-based-access-control/elevate-access-global-admin[Elevate access to manage all Azure subscriptions and management groups].

. Under *Setup Access*, select *ARM Template*.
. Under **Where to add this integration**:
.. Select **New Hosts**.
.. Name the {agent} policy. Use a name that matches the resources you want to monitor. For example, `azure-dev-policy`. Click **Save and continue**. The *ARM Template deployment* window appears.
.. In a new tab, log in to the Azure portal, then return to {kib} and click **Launch ARM Template**. This will open the ARM template in Azure.

> .. If you are deploying to an Azure organization, select the management group you want to monitor from the drop-down menu. Next, enter the subscription ID of the subscription where you want to deploy the VM that will scan your resources.

Reviewer 2: A link to the subscriptions page might be helpful: https://portal.azure.com/#view/Microsoft_Azure_Billing/SubscriptionsBladeV2

Not sure if there is any risk for the link to stop working in the future.

Author: Thanks Orestis, we usually try to avoid linking to another company's docs for exactly the reason you mentioned. I'm hoping most people will know how to find the subscription ID, or can figure it out with a quick google.

. From the Elastic Security *Get started* page, click *Add integrations*.
. Search for `CSPM`, then click on the result.
. Click *Add Cloud Security Posture Management (CSPM)*.
. Under **Configure integration**, select **Azure**, then select either **Azure Organization** or **Single Subscription**, depending on what resources you want to monitor. 
. Give your integration a name that matches the purpose or team of the Azure resources you want to monitor, for example, `azure-CSPM-dev-1`.

[discrete]
[[cspm-set-up-cloud-access-section-azure]]
=== Set up cloud account access

NOTE: To set up CSPM for an Azure organization or subscription, you will need admin privileges for that organization or subscription.

For most users, the simplest option is to use an Azure Resource Manager (ARM) template to automatically provision the necessary resources and permissions in Azure. If you prefer a more hands-on approach or require a specific configuration not supported by the ARM template, you can use one of the manual setup options described below.

[discrete]
[[cspm-set-up-ARM]]
== ARM template setup (recommended)

> NOTE: If you are deploying to an Azure organization, you need the following permissions: `Microsoft.Resources/deployments/*`, `Microsoft.Authorization/roleAssignments/write`. You also need to https://learn.microsoft.com/en-us/azure/role-based-access-control/elevate-access-global-admin[Elevate access to manage all Azure subscriptions and management groups].

Reviewer 3: ```suggestion
NOTE: If you are deploying to an Azure organization, you need the following permissions: `Microsoft.Resources/deployments/*`, `Microsoft.Authorization/roleAssignments/write`. You also need to https://learn.microsoft.com/en-us/azure/role-based-access-control/elevate-access-global-admin[elevate access to manage all Azure subscriptions and management groups].
```

. Under *Setup Access*, select *ARM Template*.
. Under **Where to add this integration**:
.. Select **New Hosts**.

> .. Name the {agent} policy. Use a name that matches the resources you want to monitor. For example, `azure-dev-policy`. Click **Save and continue**. The *ARM Template deployment* window appears.

Reviewer 3: ```suggestion
.. Name the {agent} policy. Use a name that matches the resources you want to monitor, for example, `azure-dev-policy`. Click **Save and continue**. The *ARM Template deployment* window appears.
```

. Select the `Reader` function role, assign access to **User, group, or service principal**, and select your new app.

Next, create a certificate. If you intend to use a password-protected certificate, you must use a pkcs12 certificate. Otherwise, you must use a pem certificate.

Create a pkcs12 certificate, for example:
```shell
# Create PEM file
openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes

# Create pkcs12 bundle using legacy flag (CLI will ask for export password)
openssl pkcs12 -legacy -export -out bundle.p12 -inkey key.pem -in cert.pem
```

Create a PEM certificate, for example:
```shell
# Generate certificate signing request (csr) and key
openssl req -new -newkey rsa:4096 -nodes -keyout cert.key -out cert.csr

# Generate PEM and self-sign with key
openssl x509 -req -sha256 -days 365 -in cert.csr -signkey cert.key -out signed.pem

# Create bundle
cat cert.key > bundle.pem
cat signed.pem >> bundle.pem
```

After creating your certificate:

. Return to Azure.
. Navigate to the **Certificates & secrets** menu. Select the **Certificates** tab.
. Click **Upload certificate**. 

> .. If you're using a PEM certiciate created using the example commands above, upload `signed.pem`.

Reviewer 3: ```suggestion
.. If you're using a PEM certificate that was created using the example commands above, upload `signed.pem`.
```

> .. If you're using a pkcs12 certificate created using the example commands above, upload `cert.pem`.

Reviewer 3: Typically, I opt to omit "that" whenever possible because I think it's filler. But in this case, I think adding "that" can help with the sentence flow and create a more clear relationship with the previous step. Up to you though!

```suggestion
.. If you're using a pkcs12 certificate that was created using the example commands above, upload `cert.pem`.
```

. Upload the certificate bundle to the VM where you will deploy {agent}.

> .. If you're using a PEM certificate created using the example commands above, upload `bundle.pem`.

Reviewer 3: Another optional edit.

```suggestion
.. If you're using a PEM certificate that was created using the example commands above, upload `bundle.pem`.
```

> .. If you're using a pkcs12 certificate created using the example commands above, upload `bundle.p12`.

Reviewer 3: ```suggestion
.. If you're using a pkcs12 certificate that was created using the example commands above, upload `bundle.p12`.
```

. Return to the **Add CSPM** page in {kib}. 

> . For **Client Certificate Path**, enter the fullpath to the certificate you uploaded to the host where you will install {agent}. 

Reviewer 3: I'm not sure if fullpath should be one word, so leaving an edit here in case it shouldn't. Also suggesting an optional "that" to break up the sentence a bit.

```suggestion
. For **Client Certificate Path**, enter the full path to the certificate that you uploaded to the host where you will install {agent}. 
```

> . Go to your Azure subscription list and select the subscription you want to monitor with CSPM.

Author: What if you want to monitor an organization instead of a single subscription? Should this step be different?

. Go to **Access control (IAM)** and select **Add Role Assignment**. 
. Select the `Reader` function role, assign access to **User, group, or service principal**, and select your new app.

> . Return to the **Add CSPM** page in {kib}. 

Reviewer 4: Would it be helpful to add before the step `Return to the **Add CSPM** page in {kib}.` , another step about creating the secret in App registration? 
e.g. something like:
. Go to App Registration you created for the agent, in tab  **Certificate & Secrets** and then **Client secrets** tab, and create and copy a new secret for the elastic agent.

Author: Maybe I'm missing something, but right now line 103 has something very similar to what you're suggesting here. Do users need to make a second secret at this point or is the existing instruction enough?

Reviewer 4: Yes, you are right,  my mistake. This is the step.

. Under **Configure integration**, select **Azure**. Under **Setup access**, select **Manual**.
. Under **Preferred manual method**, select **Service principal with Client Secret**.
. Fill in the values you copied earlier for **Client ID**, **Tenant ID**, and **Client Secret**.
. Under **Where to add this integration**, select **New hosts**.
. Click **Save and continue**, then follow the instructions to install {agent} on your selected host.

Wait for the confirmation that {kib} received data from your new integration. Then you can click **View Assets** to see your data.

[discrete]
[[cspm-azure-client-certificate]]
=== Option 3: Service principal with client certificate

Before using this method, you must have set up a https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in[Microsoft Entra application and service principal that can access resources]. 

> . Go to the **Registered apps** section of https://ms.portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps[Microsoft Entra ID].

Reviewer 4: Perhaps before this step we could add the same steps as the `client secret` options regarding the role assignment? 

The steps are:
```
. Go to your Azure subscription list and select the subscription you want to monitor with CSPM.
. Go to **Access control (IAM)** and select **Add Role Assignment**. 
. Select the `Reader` function role, assign access to **User, group, or service principal**, and select your new app.
```

. Click on **New Registration**, name your app and click **Register**.
. Copy your new app's `Directory (tenant) ID` and `Application (client) ID`.

> 

Reviewer 4: Perhaps the steps regarding role assignment could be added here as well? 

```
. Go to your Azure subscription list and select the subscription you want to monitor with CSPM.
. Go to **Access control (IAM)** and select **Add Role Assignment**. 
. Select the `Reader` function role, assign access to **User, group, or service principal**, and select your new app.
```
